diff --git a/modules/storm_layout_builder/css/layout.css b/modules/storm_layout_builder/css/layout.css
index 67e190a..0f4dd37 100644
--- a/modules/storm_layout_builder/css/layout.css
+++ b/modules/storm_layout_builder/css/layout.css
@@ -60,3 +60,132 @@
 .slb-bg-black {
   background-color: #000 !important;
 }
+
+.blb-padding-top-s {
+  padding-top: 8px;
+}
+
+.blb-padding-top-m {
+  padding-top: 16px;
+}
+
+.blb-padding-top-l {
+  padding-top: 24px;
+}
+
+.blb-padding-top-xl {
+  padding-top: 32px;
+}
+
+.blb-padding-bottom-s {
+  padding-bottom: 8px;
+}
+
+.blb-padding-bottom-m {
+  padding-bottom: 16px;
+}
+
+.blb-padding-bottom-l {
+  padding-bottom: 24px;
+}
+
+.blb-padding-bottom-xl {
+  padding-bottom: 32px;
+}
+
+.blb-padding-left-s {
+  padding-left: 8px;
+}
+
+.blb-padding-left-m {
+  padding-left: 16px;
+}
+
+.blb-padding-left-l {
+  padding-left: 24px;
+}
+
+.blb-padding-left-xl {
+  padding-left: 32px;
+}
+
+.blb-padding-right-s {
+  padding-right: 8px;
+}
+
+.blb-padding-right-m {
+  padding-right: 16px;
+}
+
+.blb-padding-right-l {
+  padding-right: 24px;
+}
+
+.blb-padding-right-xl {
+  padding-right: 32px;
+}
+
+.blb-spacing-top-s {
+  margin-top: 8px;
+}
+
+.blb-spacing-top-m {
+  margin-top: 16px;
+}
+
+.blb-spacing-top-l {
+  margin-top: 24px;
+}
+
+.blb-spacing-top-xl {
+  margin-top: 32px;
+}
+
+.blb-spacing-bottom-s {
+  margin-bottom: 8px;
+}
+
+.blb-spacing-bottom-m {
+  margin-bottom: 16px;
+}
+
+.blb-spacing-bottom-l {
+  margin-bottom: 24px;
+}
+
+.blb-spacing-bottom-xl {
+  margin-bottom: 32px;
+}
+
+.blb-spacing-left-s {
+  margin-left: 8px;
+}
+
+.blb-spacing-left-m {
+  margin-left: 16px;
+}
+
+.blb-spacing-left-l {
+  margin-left: 24px;
+}
+
+.blb-spacing-left-xl {
+  margin-left: 32px;
+}
+
+.blb-spacing-right-s {
+  margin-right: 8px;
+}
+
+.blb-spacing-right-m {
+  margin-right: 16px;
+}
+
+.blb-spacing-right-l {
+  margin-right: 24px;
+}
+
+.blb-spacing-right-xl {
+  margin-right: 32px;
+}
+
diff --git a/modules/storm_layout_builder/css/storm-layout-builder-settings.css b/modules/storm_layout_builder/css/storm-layout-builder-settings.css
index a9f119d..53edb0a 100644
--- a/modules/storm_layout_builder/css/storm-layout-builder-settings.css
+++ b/modules/storm_layout_builder/css/storm-layout-builder-settings.css
@@ -1,15 +1,15 @@
 /* Form elements overrides */
 
-.form-item-layout-settings-section-background-color {
+.form-item-layout-settings-section-background-background-color {
   display: inline-block !important;
   margin-right: 5px;
 }
 
-.form-item-layout-settings-section-background-color .glb-form-boolean {
+.form-item-layout-settings-section-background-background-color .glb-form-boolean {
   display: none;
 }
 
-.form-item-layout-settings-section-background-color .glb-form-item__label.glb-option {
+.form-item-layout-settings-section-background-background-color .glb-form-item__label.glb-option {
   width: 46px;
   height: 46px;
   border: 1px dotted #ccc;
@@ -17,16 +17,16 @@
   padding: 0;
 }
 
-.form-item-layout-settings-section-background-color .glb-form-boolean:checked + .glb-form-item__label.glb-option span {
+.form-item-layout-settings-section-background-background-color .glb-form-boolean:checked + .glb-form-item__label.glb-option span {
   background: url("../assets/check.png") no-repeat center center;
 }
 
-.form-item-layout-settings-section-background-color .slb-bg-none,
-.form-item-layout-settings-section-background-color .glb-form-boolean:checked + .glb-form-item__label.glb-option span.slb-bg-none {
+.form-item-layout-settings-section-background-background-color .slb-bg-none,
+.form-item-layout-settings-section-background-background-color .glb-form-boolean:checked + .glb-form-item__label.glb-option span.slb-bg-none {
   background: url("../assets/cancel.png") no-repeat center center !important;
 }
 
-.form-item-layout-settings-section-background-color .glb-form-item__label.glb-option span {
+.form-item-layout-settings-section-background-background-color .glb-form-item__label.glb-option span {
   display: block;
   width: 100%;
   height: 100%;
diff --git a/modules/storm_layout_builder/layouts/fourcol_section/layout--fourcol-section.html.twig b/modules/storm_layout_builder/layouts/fourcol_section/layout--fourcol-section.html.twig
index 363e9e9..d8afbf3 100644
--- a/modules/storm_layout_builder/layouts/fourcol_section/layout--fourcol-section.html.twig
+++ b/modules/storm_layout_builder/layouts/fourcol_section/layout--fourcol-section.html.twig
@@ -12,13 +12,13 @@
 #}
 
 {# Set the title variables #}
-{% set title = content['#settings']['heading'] %}
-{% set heading_style = content['#settings']['heading_style'] %}
+{% set title = content['#settings']['section_title']['heading'] %}
+{% set heading_style = content['#settings']['section_title']['heading_style'] %}
 
 {% if content.first or content.second or content.third or content.fourth %}
   <div{{layout_container.attributes}}>
     {% if title %}
-      <{{heading_style}} class="c-title c-title--{{ content['#settings']['heading_alignment'] }}">
+      <{{heading_style}} class="c-title c-title--{{ content['#settings']['section_title']['heading_alignment'] }}">
         {{ title }}
       </{{heading_style}}>
     {% endif %}
diff --git a/modules/storm_layout_builder/layouts/onecol_section/layout--onecol.html.twig b/modules/storm_layout_builder/layouts/onecol_section/layout--onecol.html.twig
index ca45da2..57198f2 100644
--- a/modules/storm_layout_builder/layouts/onecol_section/layout--onecol.html.twig
+++ b/modules/storm_layout_builder/layouts/onecol_section/layout--onecol.html.twig
@@ -12,13 +12,13 @@
 #}
 
 {# Set the title variables #}
-{% set title = content['#settings']['heading'] %}
-{% set heading_style = content['#settings']['heading_style'] %}
+{% set title = content['#settings']['section_title']['heading'] %}
+{% set heading_style = content['#settings']['section_title']['heading_style'] %}
 
 {% if content.content %}
   <div{{layout_container.attributes}}>
     {% if title %}
-      <{{heading_style}} class="c-title c-title--{{ content['#settings']['heading_alignment'] }}">
+      <{{heading_style}} class="c-title c-title--{{ content['#settings']['section_title']['heading_alignment'] }}">
         {{ title }}
       </{{heading_style}}>
     {% endif %}
diff --git a/modules/storm_layout_builder/layouts/threecol_section/layout--threecol-section.html.twig b/modules/storm_layout_builder/layouts/threecol_section/layout--threecol-section.html.twig
index cc53bde..f001260 100644
--- a/modules/storm_layout_builder/layouts/threecol_section/layout--threecol-section.html.twig
+++ b/modules/storm_layout_builder/layouts/threecol_section/layout--threecol-section.html.twig
@@ -12,13 +12,13 @@
 #}
 
 {# Set the title variables #}
-{% set title = content['#settings']['heading'] %}
-{% set heading_style = content['#settings']['heading_style'] %}
+{% set title = content['#settings']['section_title']['heading'] %}
+{% set heading_style = content['#settings']['section_title']['heading_style'] %}
 
 {% if content.first or content.second or content.third %}
   <div{{layout_container.attributes}}>
     {% if title %}
-      <{{heading_style}} class="c-title c-title--{{ content['#settings']['heading_alignment'] }}">
+      <{{heading_style}} class="c-title c-title--{{ content['#settings']['section_title']['heading_alignment'] }}">
         {{ title }}
       </{{heading_style}}>
     {% endif %}
diff --git a/modules/storm_layout_builder/layouts/twocol_section/layout--twocol-section.html.twig b/modules/storm_layout_builder/layouts/twocol_section/layout--twocol-section.html.twig
index c4408b1..3eae7cc 100644
--- a/modules/storm_layout_builder/layouts/twocol_section/layout--twocol-section.html.twig
+++ b/modules/storm_layout_builder/layouts/twocol_section/layout--twocol-section.html.twig
@@ -12,13 +12,13 @@
 #}
 
 {# Set the title variables #}
-{% set title = content['#settings']['heading'] %}
-{% set heading_style = content['#settings']['heading_style'] %}
+{% set title = content['#settings']['section_title']['heading'] %}
+{% set heading_style = content['#settings']['section_title']['heading_style'] %}
 
 {% if content.first or content.second %}
   <div{{layout_container.attributes}}>
     {% if title %}
-      <{{heading_style}} class="c-title c-title--{{ content['#settings']['heading_alignment'] }}">
+      <{{heading_style}} class="c-title c-title--{{ content['#settings']['section_title']['heading_alignment'] }}">
         {{ title }}
       </{{heading_style}}>
     {% endif %}
diff --git a/modules/storm_layout_builder/src/Form/StormLayoutBuilderStyleSettingsForm.php b/modules/storm_layout_builder/src/Form/StormLayoutBuilderStyleSettingsForm.php
index 0401abd..20f171d 100644
--- a/modules/storm_layout_builder/src/Form/StormLayoutBuilderStyleSettingsForm.php
+++ b/modules/storm_layout_builder/src/Form/StormLayoutBuilderStyleSettingsForm.php
@@ -40,6 +40,66 @@ class StormLayoutBuilderStyleSettingsForm extends ConfigFormBase {
       '#description' => $this->t('<p>Enter one value per line, in the format <b>key|label</b> where <em>key</em> is the CSS class name (without the .), and <em>label</em> is the human readable name of the background.</p>'),
       '#default_value' => $this->config('storm_layout_builder.settings')->get('background_colors'),
     ];
+
+    $form['padding'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Padding'),
+    ];
+
+    $form['padding']['markup'] = [
+      '#type' => 'markup',
+      '#markup' => '<p>' . $this->t('Enter one value per line, in the format <b>key|label</b> where <em>key</em> is the CSS class name (without the .), and <em>label</em> is the human readable name.') . '</p>',
+    ];
+
+    $form['padding']['padding_top'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Padding top'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('padding_top'),
+    ];
+    $form['padding']['padding_bottom'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Padding bottom'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('padding_bottom'),
+    ];
+    $form['padding']['padding_left'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Padding left'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('padding_left'),
+    ];
+    $form['padding']['padding_right'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Padding right'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('padding_right'),
+    ];
+
+    $form['spacing'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Spacing'),
+    ];
+    $form['spacing']['markup'] = [
+      '#type' => 'markup',
+      '#markup' => '<p>' . $this->t('Enter one value per line, in the format <b>key|label</b> where <em>key</em> is the CSS class name (without the .), and <em>label</em> is the human readable name.') . '</p>',
+    ];
+    $form['spacing']['spacing_top'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Spacing top'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('spacing_top'),
+    ];
+    $form['spacing']['spacing_bottom'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Spacing bottom'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('spacing_bottom'),
+    ];
+    $form['spacing']['spacing_left'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Spacing left'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('spacing_left'),
+    ];
+    $form['spacing']['spacing_right'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Spacing right'),
+      '#default_value' => $this->config('storm_layout_builder.settings')->get('spacing_right'),
+    ];
     return parent::buildForm($form, $form_state);
   }
 
@@ -47,9 +107,12 @@ class StormLayoutBuilderStyleSettingsForm extends ConfigFormBase {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $this->config('storm_layout_builder.settings')
-      ->set('background_colors', trim($form_state->getValue('background_colors')))
-      ->save();
+    $configuration = $this->config('storm_layout_builder.settings');
+    foreach ($form_state->getValues() as $key => $value) {
+      $configuration->set($key, trim($value));
+    }
+    $configuration->save();
+
     parent::submitForm($form, $form_state);
   }
 
diff --git a/modules/storm_layout_builder/src/Plugin/Layout/StormLayout.php b/modules/storm_layout_builder/src/Plugin/Layout/StormLayout.php
index 4de15d1..6bf0767 100644
--- a/modules/storm_layout_builder/src/Plugin/Layout/StormLayout.php
+++ b/modules/storm_layout_builder/src/Plugin/Layout/StormLayout.php
@@ -91,13 +91,13 @@ class StormLayout extends LayoutDefault implements ContainerFactoryPluginInterfa
         'h5' => $this->t('H5'),
         'h6' => $this->t('H6'),
       ],
-      '#default_value' => $this->configuration['heading_style'] ?? 'h1',
+      '#default_value' => $this->configuration['section_title']['heading_style'] ?? 'h1',
     ];
 
     $form['section_title']['heading'] = [
       '#type' => 'textfield',
       '#title' => $this->t('Title'),
-      '#default_value' => $this->configuration['heading'] ?? '',
+      '#default_value' => $this->configuration['section_title']['heading'] ?? '',
     ];
 
     $form['section_title']['heading_alignment'] = [
@@ -108,7 +108,7 @@ class StormLayout extends LayoutDefault implements ContainerFactoryPluginInterfa
         'text-align-right' => $this->t('Right'),
         'text-align-center' => $this->t('Center'),
       ],
-      '#default_value' => $this->configuration['heading_alignment'] ?? 'text-align-left',
+      '#default_value' => $this->configuration['section_title']['heading_alignment'] ?? 'text-align-left',
     ];
 
     $form['section_background'] = [
@@ -117,20 +117,20 @@ class StormLayout extends LayoutDefault implements ContainerFactoryPluginInterfa
       '#weight' => 3,
     ];
 
-    $form['section_background']['color'] = [
+    $form['section_background']['background_color'] = [
       '#type' => 'radios',
       '#options' => $this->getColors(),
-      '#default_value' => $this->configuration['background_color'] ?? 'slb-bg-none',
+      '#default_value' => $this->configuration['section_background']['background_color'] ?? 'slb-bg-none',
     ];
 
-    $form['section_background']['media'] = [
+    $form['section_background']['background_media'] = [
       '#type' => 'media_library',
       '#allowed_bundles' => ['image'],
       '#title' => $this->t('Background Image'),
-      '#default_value' => $this->configuration['background_media'] ?? NULL,
+      '#default_value' => $this->configuration['section_background']['background_media'] ?? NULL,
       '#description' => $this->t('Upload or select a background image.'),
     ];
-    $form['section_background']['position'] = [
+    $form['section_background']['background_position'] = [
       '#type' => 'select',
       '#title' => $this->t('Background image position'),
       '#options' => [
@@ -144,20 +144,20 @@ class StormLayout extends LayoutDefault implements ContainerFactoryPluginInterfa
         'right center' => $this->t('Right center'),
         'right bottom' => $this->t('Right bottom'),
       ],
-      '#default_value' => $this->configuration['background_position'] ?? 'center center',
+      '#default_value' => $this->configuration['section_background']['background_position'] ?? 'center center',
     ];
 
-    $form['section_background']['size'] = [
+    $form['section_background']['background_size'] = [
       '#type' => 'select',
       '#title' => $this->t('Background image size'),
       '#options' => [
         'cover' => $this->t('Cover'),
         'contain' => $this->t('Contain'),
       ],
-      '#default_value' => $this->configuration['background_size'] ?? 'cover',
+      '#default_value' => $this->configuration['section_background']['background_size'] ?? 'cover',
     ];
 
-    $form['section_background']['repeat'] = [
+    $form['section_background']['background_repeat'] = [
       '#type' => 'select',
       '#title' => $this->t('Background image repeat'),
       '#options' => [
@@ -166,9 +166,61 @@ class StormLayout extends LayoutDefault implements ContainerFactoryPluginInterfa
         'repeat-x' => $this->t('Repeat X'),
         'repeat-y' => $this->t('Repeat Y'),
       ],
-      '#default_value' => $this->configuration['background_repeat'] ?? 'no-repeat',
+      '#default_value' => $this->configuration['section_background']['background_repeat'] ?? 'no-repeat',
     ];
 
+    $form['section_padding'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Padding'),
+      '#weight' => 3,
+    ];
+
+    $padding = [
+      'padding_top' => $this->t('Top padding'),
+      'padding_bottom' => $this->t('Bottom padding'),
+      'padding_left' => $this->t('Left padding'),
+      'padding_right' => $this->t('Right padding'),
+    ];
+
+    foreach ($padding as $key => $value) {
+      $form['section_padding'][$key] = [
+        '#type' => 'select',
+        '#title' => $value,
+        '#options' => [
+          'none' => $this->t('None'),
+        ],
+        '#empty_value' => 'none',
+        '#default_value' => $this->configuration['section_padding'][$key] ?? 'none',
+      ];
+      $form['section_padding'][$key]['#options'] = $this->getSpacings($key);
+    }
+
+    $form['section_spacing'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Spacing'),
+      '#weight' => 3,
+    ];
+
+    $spacing = [
+      'spacing_top' => $this->t('Top spacing'),
+      'spacing_bottom' => $this->t('Bottom spacing'),
+      'spacing_left' => $this->t('Left spacing'),
+      'spacing_right' => $this->t('Right spacing'),
+    ];
+
+    foreach ($spacing as $key => $value) {
+      $form['section_spacing'][$key] = [
+        '#type' => 'select',
+        '#title' => $value,
+        '#options' => [
+          'none' => $this->t('None'),
+        ],
+        '#empty_value' => 'none',
+        '#default_value' => $this->configuration['section_spacing'][$key] ?? 'none',
+      ];
+      $form['section_spacing'][$key]['#options'] = $this->getSpacings($key);
+    }
+
     return parent::buildConfigurationForm($form, $form_state);
   }
 
@@ -188,6 +240,14 @@ class StormLayout extends LayoutDefault implements ContainerFactoryPluginInterfa
     return $options;
   }
 
+  /**
+   * Helper method to get padding options.
+   */
+  private function getSpacings($key) {
+    $config = $this->configFactory->get('storm_layout_builder.settings')->get($key);
+    return $this->getConfigValues($config);
+  }
+
   /**
    * Build the config values array from config object.
    */
@@ -208,17 +268,10 @@ class StormLayout extends LayoutDefault implements ContainerFactoryPluginInterfa
    */
   public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
     parent::submitConfigurationForm($form, $form_state);
-    $section_title = $form_state->getValue('section_title');
-    $this->configuration['heading'] = $section_title['heading'];
-    $this->configuration['heading_style'] = $section_title['heading_style'];
-    $this->configuration['heading_alignment'] = $section_title['heading_alignment'];
-
-    $background = $form_state->getValue('section_background');
-    $this->configuration['background_color'] = $background['color'];
-    $this->configuration['background_media'] = $background['media'] ?? NULL;
-    $this->configuration['background_position'] = $background['position'];
-    $this->configuration['background_size'] = $background['size'];
-    $this->configuration['background_repeat'] = $background['repeat'];
+
+    foreach ($form_state->getValues() as $key => $config) {
+      $this->configuration[$key] = $config;
+    }
   }
 
 }
diff --git a/modules/storm_layout_builder/storm_layout_builder.info.yml b/modules/storm_layout_builder/storm_layout_builder.info.yml
index dbf308e..18de528 100644
--- a/modules/storm_layout_builder/storm_layout_builder.info.yml
+++ b/modules/storm_layout_builder/storm_layout_builder.info.yml
@@ -14,5 +14,4 @@ dependencies:
   - layout_builder_restrictions:layout_builder_restrictions
   - layout_disable:layout_disable
   - layout_library:layout_library
-  - media_library_form_element:media_library_form_element
 
diff --git a/modules/storm_layout_builder/storm_layout_builder.module b/modules/storm_layout_builder/storm_layout_builder.module
index 2e7c991..59add78 100644
--- a/modules/storm_layout_builder/storm_layout_builder.module
+++ b/modules/storm_layout_builder/storm_layout_builder.module
@@ -20,10 +20,39 @@ function storm_layout_builder_plugin_filter_layout__layout_builder_alter(array &
  * Implements hook_preprocess_layout().
  */
 function storm_layout_builder_preprocess_layout(&$variables) {
-  $media = $variables['settings']['background_media'] ?? NULL;
-  $position = $variables['settings']['background_position'] ?? NULL;
-  $size = $variables['settings']['background_size'] ?? NULL;
-  $repeat = $variables['settings']['background_repeat'] ?? NULL;
+  // Build the Layout container class.
+  switch ($variables['content']['#layout']->id()) {
+    case 'storm_layout_onecol':
+      $layout_modifier = 'l-container--onecol';
+      break;
+
+    case 'storm_layout_twocol_section':
+      $layout_modifier = 'l-container--twocol';
+      break;
+
+    case 'storm_layout_threecol_section':
+      $layout_modifier = 'l-container--threecol';
+      break;
+
+    case 'storm_layout_fourcol_section':
+      $layout_modifier = 'l-container--fourcol';
+      break;
+  }
+
+  $class = [
+    'l-container',
+    $layout_modifier,
+  ];
+
+  // Merge all modifier classes.
+  $class = array_merge($class, [$variables['settings']['section_background']['background_color']], $variables['settings']['section_padding'], $variables['settings']['section_spacing']);
+  // Filter out any none values.
+  $class = array_filter($class, fn ($m) => $m != 'none');
+
+  $media = $variables['settings']['section_background']['background_media'] ?? NULL;
+  $position = $variables['settings']['section_background']['background_position'] ?? NULL;
+  $size = $variables['settings']['section_background']['background_size'] ?? NULL;
+  $repeat = $variables['settings']['section_background']['background_repeat'] ?? NULL;
 
   if ($media) {
     $entity = Media::load($media);
@@ -41,30 +70,8 @@ function storm_layout_builder_preprocess_layout(&$variables) {
     }
   }
 
-  switch ($variables['content']['#layout']->id()) {
-    case 'storm_layout_onecol':
-      $class = 'l-container--onecol';
-      break;
-
-    case 'storm_layout_twocol_section':
-      $class = 'l-container--twocol';
-      break;
-
-    case 'storm_layout_threecol_section':
-      $class = 'l-container--threecol';
-      break;
-
-    case 'storm_layout_fourcol_section':
-      $class = 'l-container--fourcol';
-      break;
-  }
-
   $variables['layout_container']['attributes'] = new Attribute([
-    'class' => [
-      'l-container',
-      $class,
-      $variables['settings']['background_color'],
-    ],
+    'class' => $class,
     'style' => $background_media ?? NULL,
   ]);
 }
